node/src/pow/blake3_pow.rs

```diff
@@ -0,0 +1,137 @@
+// Radicle Registry
+// Copyright (C) 2019 Monadic GmbH <radicle@monadic.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! PoW algorithm implementation based on Blake3 hashing
+//!
+//! The difficulty is an average number of hashes that need to be checked in order mine a block.
+//! It must be capped at `2^192 - 1`. This limit guarantees that Substrate can calculate a sum
+//! of difficulties of blocks in a chain of up to `2^64` height and not overflow. This cap
+//! should be impossible to break with current technology due to constraints of the universe.
+//!
+//! In order to check if the data hash passes the difficulty test, it must be interpreted as
+//! a big-endian 256-bit number. If it's smaller than or equal to the threshold, it passes.
+//! The threshold is calculated from difficulty as `U256::max_value / difficulty`.
+//!
+//! There's no difficulty adjustment algorithm yet.
+
+use radicle_registry_runtime::opaque::Block;
+use radicle_registry_runtime::Hash;
+use sc_consensus_pow::{Error, PowAlgorithm};
+use sp_consensus_pow::Seal;
+use sp_core::uint::U256;
+use sp_runtime::generic::BlockId;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+type Difficulty = U256;
+type Threshold = U256;
+
+const NONCES_PER_MINING_ROUND: usize = 10_000_000;
+
+/// The implementor of the Blake3 PoW algorithm.
+///
+/// For more information about this PoW algorithm see [module](index.html) documentation.
```

Comment:
I tried running `cargo doc` to see these as docs but for some reason, I don't see it as part of the targe/doc output.
That said, two things:

1. 
```suggestion
/// For more information about this PoW algorithm see the [module](index.html) documentation.
```

2. The `[module](index.html)` seems to point to nowhere? 🤔

Reply:
You can run `cargo doc --open` to open it in the browser, you can then use the regular search to find this item.

The structure has address `struct.Blake3Pow.html` and the module has just `index.html`. I don't like it (assumption of HTML output), but that seems to be the right way to create internal links in rustdoc :man_shrugging: